### PR TITLE
Services refactor (remove bluebird dependency)

### PR DESCRIPTION
**Pull request by pablopareja**
## Overview
This PR includes the following updates:
- New area service
- New subscriptions service
- User service refactor (bluebird dependency removed)

## [Pivotal task](https://www.pivotaltracker.com/story/show/165015958)

### CODE REVIEW

File: components/areas/AlertWidget.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ import AreaSubscriptionModal from 'components/modal/AreaSubscriptionModal';
 import LayerInfoModal from 'components/modal/layer-info-modal';
 
 // Services
-import AreasService from 'services/AreasService';
+import { getGeostore } from 'services/areas';
```
Comment (andresgnlez) [inline components/areas/AlertWidget.js]: Geostore operations should have its own service.

File: components/areas/AreasForm.js [hunk 2]
```diff
@@ -16,8 +16,8 @@ import Input from 'components/form/Input';
 import UploadArea from 'components/areas/UploadArea';
 
 // Services
-import AreasService from 'services/AreasService';
-import UserService from 'services/user';
+import { fetchCountries, createGeostore } from 'services/areas';
```
Comment (andresgnlez) [inline components/areas/AreasForm.js]: https://github.com/resource-watch/resource-watch/pull/1116/files#r324716188 <- same. The services have a unique purpose.

File: components/subscriptions/SubscriptionCard.js [hunk 3]
```diff
@@ -12,8 +12,8 @@ import Map from 'components/widgets/map/Map';
 
 // Services
 import DatasetService from 'services/DatasetService';
-import AreasService from 'services/AreasService';
-import UserService from 'services/user';
+import { getCountry, getGeostore } from 'services/areas';
```
Comment (andresgnlez) [inline components/subscriptions/SubscriptionCard.js]: Same. Why the area service would get countries or geostores?

File: redactions/user.js [hunk 4]
```diff
@@ -9,11 +9,25 @@ import { mergeSubscriptions, setGeoLayer, setCountryLayer } from 'utils/user/are
 import { getDatasetsByTab } from 'redactions/admin/datasets';
 
 // services
-import UserService from 'services/user';
-import FavouritesService from 'services/favourites';
+import {
+  getUserAreas as getUserAreasService,
+  deleteArea
+} from 'services/user';
```
Comment (andresgnlez) [inline redactions/user.js]: Same. No sense user service gets or deletes areas.

File: redactions/user.js [hunk 5]
```diff
@@ -9,11 +9,25 @@ import { mergeSubscriptions, setGeoLayer, setCountryLayer } from 'utils/user/are
...
+  getSubscriptions,
+  deleteSubscription
+} from 'services/subscriptions';
 import CollectionsService from 'services/collections';
 import DatasetService from 'services/DatasetService';
-import AreasService from 'services/AreasService';
+import {
+  getCountry,
+  getGeostore
+} from 'services/areas';
```
Comment (andresgnlez) [inline redactions/user.js]: https://github.com/resource-watch/resource-watch/pull/1116/files#r324718389

File: services/areas.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    geojson,
+    { headers: { 'Content-Type': 'application/json' } })
+    .catch(() => { throw new Error("The file couldn't be processed correctly. Try again in a few minutes."); });
+  return response.data.data;
+};
+
+/**
+ * Get area
+ */
+export const getArea = (id, token) =>
```
Comment (andresgnlez) [inline services/areas.js]: For consistency, fetches are named `fetchX` and the methods you use them are usually named `getX`.

File: services/areas.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+          return 1;
+        }
+        // eslint-disable-line no-else-return
+        return 0;
+      }));
+
+/**
+ * Get country
+ */
+export const getCountry = iso =>
```
Comment (andresgnlez) [inline services/areas.js]: Use query endpoint instead. This function has nothing to do here.

File: services/areas.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  WRIAPI.get(`query/134caa0a-21f7-451d-a7fe-30db31a424aa?sql=SELECT name_engli as label, st_asgeojson(the_geom_simple) as geojson, bbox as bounds from gadm28_countries WHERE iso = '${iso}'`);
+
+/**
+ * Get Geostore
+ */
+export const getGeostore = id => WRIAPI.get(`geostore/${id}`);
+
+export const createGeostore = async (geojson) => {
```
Comment (andresgnlez) [inline services/areas.js]: CRUD operations always return promises.

File: services/areas.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+  const response = await WRIAPI.post('geostore',
+    geojson,
+    { headers: { 'Content-Type': 'application/json' } })
```
Comment (andresgnlez) [inline services/areas.js]: `'Content-Type': 'application/json'` No need. This is applied by axios by default.

File: services/areas.js [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+    .catch(() => { throw new Error("The file couldn't be processed correctly. Try again in a few minutes."); });
+  return response.data.data;
+};
+
+/**
+ * Get area
+ */
+export const getArea = (id, token) =>
+  WRIAPI.get(
+    `area/${id}?application=${process.env.APPLICATIONS}&env=${process.env.API_ENV}`,
```
Comment (andresgnlez) [inline services/areas.js]: Too custom, allow params, never assume the behavior.

File: services/areas.js [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+    {
+      headers: {
+        'Content-Type': 'application/json',
```
Comment (andresgnlez) [inline services/areas.js]: No need.

File: services/favourites.js [hunk 12]
```diff
@@ -1,110 +1,57 @@
...
-  static getFavourites(token) {
-    const queryParams = queryString.stringify({
+import { WRIAPI } from 'utils/axios';
+
+/**
+ * Retrieve all favourites items of the user
+ * @param {*} token User's token
+ */
+export const getFavourites = (token) => {
+  const queryParams = queryString.stringify({
```
Comment (andresgnlez) [inline services/favourites.js]: with `axios`, `queryString ` is no need anymore.

File: services/favourites.js [hunk 13]
```diff
@@ -1,110 +1,57 @@
...
+    .then(response => response.data);
+};
+
+/**
+ * Creates a new favourite item attached to the current user
+ * @param {*} token User's token
+ * @param {*} resourceId Id of the resource
+ * @param {*} resourceType Resource's type (can be dataset, layer or widget)
+ */
+export const createFavourite = (token, { resourceId, resourceType }) =>
```
Comment (andresgnlez) [inline services/favourites.js]: Allow params, do not assume them.

File: services/favourites.js [hunk 14]
```diff
@@ -1,110 +1,57 @@
...
-          throw errorObject;
-        })
-        .then(data => resolve(data))
-        .catch((errors) => { reject(errors); });
+      resourceId,
+      resourceType
+    },
+    {
+      headers: {
+        'Content-Type': 'application/json',
```
Comment (andresgnlez) [inline services/favourites.js]: No need.

File: services/favourites.js [hunk 15]
```diff
@@ -1,110 +1,57 @@
...
-  }
-}
 
-export default FavouritesService;
+/**
+ * Deletes an existing favourite item attached to the current user
+ * @param {*} token User's token
+ * @param {*} resourceId Id of the resource
+ */
+export const deleteFavourite = (token, resourceId) =>
```
Comment (andresgnlez) [inline services/favourites.js]: Add error handling as we do in other services. All services must log and throw errors (to console)

File: services/favourites.js [hunk 16]
```diff
@@ -1,110 +1,57 @@
...
+  WRIAPI.delete(`/favourite/${resourceId}`,
+    { headers: { Authorization: token } });
```
Comment (andresgnlez) [inline services/favourites.js]: You're overriding defaults headers.

File: services/subscriptions.js [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+  WRIAPI.get(`subscriptions?application=${process.env.APPLICATIONS}&env=${process.env.API_ENV}`, { headers: { Authorization: token } }).then(response => response.data.data);
+
+/**
+ * Creates a subscription for a pair of dataset and country
+ * @param {datasetID} ID of the dataset
+ * @param {object} Either { type; 'iso', id:'ESP' } or { type: 'geostore', id: 'sakldfa7ads0ka'}
+ * @param {string} language Two-letter locale
+ * @returns {Promise}
+ */
+export const createSubscriptionToArea = (
```
Comment (andresgnlez) [inline services/subscriptions.js]: Make it way more simple passing params ready to be sent. Services don't care about your stuff, they just trigger fetches according to your params.

File: services/subscriptions.js [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+    resource: {
+      type: 'EMAIL',
+      content: user.email
+    },
+    params: { area: areaId }
+  };
+
+  return WRIAPI.post('subscriptions', bodyObj, {
+    headers: {
+      'Content-Type': 'application/json',
```
Comment (andresgnlez) [inline services/subscriptions.js]: No need, use axios defaults.

File: services/subscriptions.js [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+      Authorization: user.token
+    }
+  });
+};
+
+/**
+ *  Update Subscription
+ */
+export const updateSubscriptionToArea = (
```
Comment (andresgnlez) [inline services/subscriptions.js]: https://github.com/resource-watch/resource-watch/pull/1116/files#r324724138

**Comment by andresgnlez** [comment]
@pablopareja I've done a superficial  code review, check comments:

Missing: better error handling and logging. Any fetch triggered by the API must be tracked.

File: components/areas/AlertWidget.js (hunk 1 above)
Comment (pablopareja) [reply components/areas/AlertWidget.js]: done

File: services/favourites.js (hunk 13 above)
Comment (pablopareja) [reply services/favourites.js]: What do you mean?

**Pull request merged**